bk-api/bk_api.py

```diff
@@ -1316,102 +1253,95 @@ def get(self):
+        logger.info("registration request: %r", request.args)
+
+        node_id = request.args.get("node_id", type=str)
+
         if not node_id:
-            logger.debug("Registration failed: id missing")
-            return f"Error: id missing\n", 500
+            return "error: node id must be provided.\n", HTTPStatus.BAD_REQUEST
 
-        logger.debug("Register user [{}]".format(node_id))
+        if not valid_node_id(node_id):
+            return "error: invalid node_id. (node ids must be between 6 and 32 [a-z0-9] characters.)\n", HTTPStatus.BAD_REQUEST
 
+        beehive_id = request.args.get("beehive_id", DEFAULT_BEEHIVE, type=str)
 
-        if DEFAULT_BEEHIVE:
+        if beehive_id:
             # first check the default beehive exists
             try:
                 bee_db = BeekeeperDB()
-                beehive_obj = bee_db.get_beehive(DEFAULT_BEEHIVE)
+                beehive_obj = bee_db.get_beehive(beehive_id)
                 bee_db.close()
             except Exception as e:
-                raise Exception(f"error: get_beehive returned: {str(e)}")
+                raise Exception(f"error: get_beehive returned: {e}")
 
             if not beehive_obj:
-                raise Exception(f"error: Beehive {DEFAULT_BEEHIVE} is not known yet" )
+                logger.error("registration error: beehive does not exist: %r", request.args)
+                return "error: beehive not found", HTTPStatus.NOT_FOUND
 
         try:
             # create keypair and certificates for node (idempotent function)
-            registration_result =  _register(node_id)
+            registration_result = _register(node_id)
         except Exception as e:
-            logger.debug(f"_register failed: {str(e)}")
-            traceback.print_exc()
-            logger.debug(f"Error: unable to register id [{node_id} , {str(e)}]")
-            return f"Error: unable to register id [{node_id} , {str(e)}]\n", 500
+            logger.exception("registration error: _register: %r", request.args)
+            return f"error: unable to register id [{node_id} , {e}]\n", HTTPStatus.INTERNAL_SERVER_ERROR
 
         # update beekeeper db (create registartion event)
         try:
             register_node(node_id, lock_requested_by="register-resource")
         except Exception as e:
-            logger.debug(f"Error: Creating registration event failed: {str(e)}")
-            return f"Error: Creating registration event failed: {str(e)}", 500
+            logger.exception("registration error: register_node: %r", request.args)
+            return f"Error: Creating registration event failed: {str(e)}", HTTPStatus.INTERNAL_SERVER_ERROR
 
-
-        if DEFAULT_BEEHIVE:
+        if beehive_id:
             time.sleep(2) # see if that helps with the locks
-            logger.debug("Adding user [{}] to default beehive".format(node_id))
+            # TODO(sean) Review how locks are used. I'm concerned that a sleep is supposed to "help with the locks".
```

Comment:
include this in your future JIRA for fixing some of these TODO items

Reply:
Done. https://jira.cels.anl.gov/browse/SAGE-1420